File: ios-cuantrack/CuanTrack/Views/Components/SummaryCard.swift
//
//  SummaryCard.swift
//  CuanTrack
//

import SwiftUI

struct SummaryCard: View {
    let totalBalance: Double
    let monthlyExpense: Double
    let monthlyIncome: Double

    var body: some View {
        VStack(spacing: 0) {
            VStack(spacing: 6) {
                Text("Total Saldo")
                    .font(.system(size: 12))
                    .foregroundStyle(.white.opacity(0.8))

                Text("Rp \(formatAmount(totalBalance))")
                    .font(.system(size: 28, weight: .bold))
                    .foregroundStyle(.white)
            }
            .padding(.vertical, 24)
            .frame(maxWidth: .infinity)
            .background(Color(red: 0.15, green: 0.39, blue: 0.92))

            HStack(spacing: 0) {
                VStack(spacing: 4) {
                    Text("Pengeluaran")
                        .font(.system(size: 12))
                        .foregroundStyle(.white.opacity(0.8))

                    Text("Rp \(formatAmount(monthlyExpense))")
                        .font(.system(size: 16, weight: .semibold))
                        .foregroundStyle(Color(red: 1, green: 0.60, blue: 0.55))
                }
                .frame(maxWidth: .infinity)
                .padding(.vertical, 16)

                Rectangle()
                    .fill(.white.opacity(0.2))
                    .frame(width: 1, height: 32)

                VStack(spacing: 4) {
                    Text("Pemasukan")
                        .font(.system(size: 12))
                        .foregroundStyle(.white.opacity(0.8))

                    Text("Rp \(formatAmount(monthlyIncome))")
                        .font(.system(size: 16, weight: .semibold))
                        .foregroundStyle(Color(red: 0.40, green: 0.90, blue: 0.55))
                }
                .frame(maxWidth: .infinity)
                .padding(.vertical, 16)
            }
            .background(Color(red: 0.10, green: 0.31, blue: 0.78))
        }
        .clipShape(.rect(cornerRadius: 18))
        .shadow(color: Color(red: 0.15, green: 0.39, blue: 0.92).opacity(0.25), radius: 12, y: 6)
    }

    private func formatAmount(_ value: Double) -> String {
        let formatter = NumberFormatter()
        formatter.numberStyle = .decimal
        formatter.groupingSeparator = "."
        formatter.maximumFractionDigits = 0
        return formatter.string(from: NSNumber(value: value)) ?? "0"
    }
}

#Preview {
    SummaryCard(totalBalance: 12_450_000, monthlyExpense: 3_200_000, monthlyIncome: 8_500_000)
        .padding(24)
        .background(Color(red: 0.97, green: 0.98, blue: 0.99))
}

File: android-cuantrack/app/src/main/java/com/rork/cuantrackandroid/ui/components/SummaryCard.kt
package com.rork.cuantrackandroid.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.rork.cuantrackandroid.ui.theme.CuanColors

@Composable
fun SummaryCard(
    totalBalance: Double,
    monthlyExpense: Double,
    monthlyIncome: Double,
    modifier: Modifier = Modifier
) {
    val nf = rememberNumberFormatter()

    Column(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(18.dp))
            .shadow(12.dp, RoundedCornerShape(18.dp), ambientColor = CuanColors.Primary.copy(alpha = 0.25f))
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .background(CuanColors.Primary)
                .padding(vertical = 24.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Text(
                "Total Saldo",
                fontSize = 12.sp,
                color = Color.White.copy(alpha = 0.8f)
            )
            Spacer(Modifier.height(6.dp))
            Text(
                "Rp ${nf.format(totalBalance)}",
                fontSize = 28.sp,
                fontWeight = FontWeight.Bold,
                color = Color.White
            )
        }

        Row(
            modifier = Modifier
                .fillMaxWidth()
                .background(CuanColors.PrimaryDark),
        ) {
            Column(
                modifier = Modifier
                    .weight(1f)
                    .padding(vertical = 16.dp),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                Text("Pengeluaran", fontSize = 12.sp, color = Color.White.copy(alpha = 0.8f))
                Spacer(Modifier.height(4.dp))
                Text(
                    "Rp ${nf.format(monthlyExpense)}",
                    fontSize = 16.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = Color(0xFFFF9999)
                )
            }

            Box(
                modifier = Modifier
                    .padding(vertical = 8.dp)
                    .width(1.dp)
                    .height(32.dp)
                    .background(Color.White.copy(alpha = 0.2f))
            )

            Column(
                modifier = Modifier
                    .weight(1f)
                    .padding(vertical = 16.dp),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                Text("Pemasukan", fontSize = 12.sp, color = Color.White.copy(alpha = 0.8f))
                Spacer(Modifier.height(4.dp))
                Text(
                    "Rp ${nf.format(monthlyIncome)}",
                    fontSize = 16.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = Color(0xFF66E58C)
                )
            }
        }
    }
}

private fun rememberNumberFormatter(): java.text.NumberFormat {
    return java.text.NumberFormat.getNumberInstance(java.util.Locale("id", "ID")).apply {
        maximumFractionDigits = 0
    }
}
